ledger/participant-integration-api/src/main/scala/platform/store/dao/events/SqlFunctions.scala

```diff
@@ -63,14 +63,20 @@ private[dao] object SqlFunctions {
     // TODO https://github.com/digital-asset/daml/issues/9493
     // This is likely extremely inefficient due to the multiple full tablescans on unindexed varray column
     override def arrayIntersectionWhereClause(arrayColumn: String, parties: Set[Party]): String =
-      parties
-        .map(party => s"('$party') IN (SELECT * FROM TABLE($arrayColumn))")
-        .mkString("(", " or ", ")")
+      s"""JSON_EXISTS($arrayColumn, '$$[*]?(@ in ("${parties.mkString("\", \"")}"))')"""
```

Comment:
i guess the escaping is inevitable here but would be nice if we could wrap the entire thing in triple quoted strings to avoid having to escape?

Reply:
Modified it in upcoming commit but I'm not sure I made it any nicer. I'm trying to imagine nicer encodings that aren't a DSL and coming up short. Quoting quotes is an ugly affair.